### PR TITLE
Fix/1.10.x ros3

### PR DESCRIPTION
**Pull request by satra**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
-{% set version = hdf5_version|default("1.10.5") %}
+{% set version = "1.10.6" %}
```
Comment (beckermr) [inline recipe/meta.yaml:1]: Do not change this line. This branch builds both 1.10.5 and 1.10.6

**Comment by satra** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 
 {% set maj_min_ver = ".".join(version.split(".")[:2]) %}
-{% set build = 1011 %}
+{% set build = 0 %}
```
Comment (beckermr) [inline recipe/meta.yaml:4]: You need to put the build number to 1012.

**Review by beckermr — CHANGES_REQUESTED** [review]
You'll need to add code to select the flags/patches you want only on the 1.10.6 build.

Alternatively, you can wait until the 1.10.5 to 1.10.6 migration is done and then only build 1.10.6 on this branch.

**Comment by satra** [comment]
this follows up on #134 to enable the patch for 1.10.x branch. on my local machine i checked the diff, and the same patch is applicable there. so i cherrypicked most of @djhoese commits.

File: recipe/meta.yaml (hunk 1 above)
Comment (satra) [reply recipe/meta.yaml:1]: ah, the ros3 driver is only available from 1.10.6 onwards. how would i address that given the changes.

Comment (beckermr) [reply recipe/meta.yaml:1]: You can use jinja2 if blocks or selectors on the version

**Comment by satra** [comment]
@beckermr - i changed meta.yaml to respect version, but is there an example of checking for the version in build.sh and install.bat? the configure flags need to be adjusted.

File: recipe/meta.yaml (hunk 1 above)
Comment (satra) [reply recipe/meta.yaml]: resolved this in meta, but not in install.bat or build.sh

**Comment by beckermr** [comment]
The same variables in the meta.yaml selectors should be in the bash in build.sh

**Comment by satra** [comment]
@beckermr - now i am a little confused. this line for master enables the ros3 driver, but this driver is only available as of 1.10.6 (see: https://www.hdfgroup.org/2019/12/release-of-hdf5-1-10-6-newsletter-170/)

https://github.com/conda-forge/hdf5-feedstock/blob/1.10.x/recipe/build.sh#L40

i don't know whether i should be fixing this, or simply wait for the tests. i'll have to download 1.10.5 code to check if they already had that driver in there or configure simply ignores the flag.

**Comment by djhoese** [comment]
Looks like someone copied my commits from master and applied them to the 1.10 branch where they may or may not be used. If they don't error out it is hard to tell.

**Comment by satra** [comment]
thanks @djhoese - i have fixed the ros3 check and requirements to be specific to 1.10.6, but indeed the builds seem to be working fine on all platforms.  (there is one error, but that was an http timeout)

**Comment by beckermr** [comment]
If the tests of the functionality pass (and do double check they are executed), then I'd say we are good to go.

**Comment by satra** [comment]
@conda-forge-admin, please restart ci

**Comment by satra** [comment]
@beckermr - this is ready for any final reviews and if all looks good, the ros3 driver tests pass with 1.10.6

**Pull request merged**